api/service/legacysync/syncing.go

```diff
@@ -1050,10 +1050,15 @@ func (ss *StateSync) GetMaxPeerHeight() uint64 {
 }
 
 // SyncLoop will keep syncing with peers until catches up
-func (ss *StateSync) SyncLoop(bc core.BlockChain, worker *worker.Worker, isBeacon bool, consensus *consensus.Consensus) {
+func (ss *StateSync) SyncLoop(bc core.BlockChain, worker *worker.Worker, isBeacon bool, consensus *consensus.Consensus, loopMinTime time.Duration) {
+	utils.Logger().Info().Msgf("legacy sync is executing ...")
 	if !isBeacon {
 		ss.RegisterNodeInfo()
 	}
+
+	startTime := time.Now()
+	startHead := bc.CurrentBlock().NumberU64()
+	fmt.Print("\033[s") // save the cursor position
```

Comment:
remember to remove the debug prints.

Reply:
This line prints out a line like this:
**sync blocks progress: 40651 / 41670 ( 69.85 blocks/s )**
and update same line in console logs. So, it's like progress counter effect. It saves the cursor position to be able to print in same line in console. The **\033[s** is a special command to store console cursor position.

Reply:
ok